src/Bridges/Constraint/soc_to_nonconvex_quad.jl

```diff
@@ -0,0 +1,191 @@
+abstract type AbstractSOCtoNonConvexQuadBridge{T} <: AbstractBridge end
+
+"""
+    SOCtoNonConvexQuadBridge{T}
+
+Constraints of the form `VectorOfVariables`-in-`SecondOrderCone` can be
+transformed into a `ScalarQuadraticFunction`-in-`LessThan` and a
+`ScalarAffineFunction`-in-`GreaterThan`. Indeed, the definition of the
+second-order cone
+```math
+t \\ge || x ||_2 \\  (1)
```

Comment:
Do the (1), (2), (3) labels display in the REPL? (Is it correct markdown?)

Reply:
Will check the best way to do so.

Reply:
it is not proper markdown, but it renders as:
```
help?> MathOptInterface.Bridges.Constraint.RSOCtoNonConvexQuadBridge
  RSOCtoNonConvexQuadBridge{T}

  Constraints of the form VectorOfVariables-in-SecondOrderCone can be transformed into a
  ScalarQuadraticFunction-in-LessThan and a ScalarAffineFunction-in-GreaterThan. Indeed, the definition of the
  second-order cone

2tu \ge \lVert x \rVert_2^2, t,u \ge 0  (1)

  is equivalent to

\sum x_i^2 \le 2tu  (2)

  with t,u \ge 0. (3)

  WARNING This transformation starts from a convex constraint (1) and creates a non-convex constraint (2), because the
  Q matrix associated with the constraint 2 has two negative eigenvalues. This might be wrongly interpreted by a
  solver. Some solvers can look at (2) and understand that it is a rotated second order cone, but this is not a
  general rule. For these reasons this bridge is not automatically added to the LazyBridgeOptimizer. Care is
  recommended when adding this bridge to a optimizer.
```

Which is not bad in my opinion.